Source/Core/VideoBackends/OGL/OGLBoundingBox.cpp

```diff
@@ -36,43 +50,101 @@ void BoundingBox::Shutdown()
   glDeleteBuffers(1, &s_bbox_buffer_id);
 }
 
-void BoundingBox::Set(int index, int value)
+void BoundingBox::Flush()
```

Comment:
I don't know enough about BoundingBox stuff to know if this is worth it but longer term it makes me wonder if we could split some of this out to a `AbstractBoundingBox` class that does some of this common logic and calls into the backend specific functionality.  This is near identical to the `D3D11::Flush()` .  I haven't looked at the other backends to know if they are similar but you mentioned there is caching there too..